sdk/test/tutorials/clientBuilder.ts

```diff
@@ -1,10 +1,27 @@
 import { Client, Source } from '../../client/client'
 
+/*
+  Read from Environment Variables
+*/
+
+// export var client = new Client(
+//   [Source.Environment, 'FBN_TOKEN_URL'],
+//   [Source.Environment, 'FBN_USERNAME'],
+//   [Source.Environment, 'FBN_PASSWORD'],
+//   [Source.Environment, 'FBN_CLIENT_ID'],
+//   [Source.Environment, 'FBN_CLIENT_SECRET'],
+//   [Source.Environment, 'FBN_LUSID_API_URL'],
+// )
+
+/*
+  Read from secrets.json
+*/
+
 export var client = new Client(
-  [Source.Environment, 'FBN_TOKEN_URL'],
-  [Source.Environment, 'FBN_USERNAME'],
-  [Source.Environment, 'FBN_PASSWORD'],
-  [Source.Environment, 'FBN_CLIENT_ID'],
-  [Source.Environment, 'FBN_CLIENT_SECRET'],
-  [Source.Environment, 'FBN_LUSID_API_URL'],
-)
+  [Source.Secrets, 'tokenUrl'],
+  [Source.Secrets, 'username'],
+  [Source.Secrets, 'password'],
+  [Source.Secrets, 'clientId'],
+  [Source.Secrets, 'clientSecret'],
+  [Source.Secrets, 'apiUrl']
+)
```

Comment:
As this client runs the tests we need some way to ensure that we can run them locally using a secrets file and in our continuous integration pipeline which will use environment variables. 

The logic in the Python SDK is to use a secrets file if provided, falling back to environment variables if it can't find a secret file. Do you think we could implement similar logic here?

https://github.com/finbourne/lusid-sdk-python-preview/tree/master/sdk/lusid/utilities

I would suggest that this lives in the sdk/client/client.ts but then you can delete completely the commented out code which references environment variables.

Reply:
yep. will do.